File: commons/src/main/java/com/nerdscorner/mvplib/commons/mvp/presenter/Presenter.java
package com.nerdscorner.mvplib.commons.mvp.presenter;

import android.content.Intent;
import android.content.res.Configuration;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.view.MenuItem;

public interface Presenter {

    void onStart();

    void onResume();

    void onPause();

    void onStop();

    void onDestroyView();

    boolean onBackPressed();

    void onConfigurationChanged(Configuration newConfig);

    void onSaveInstanceState(Bundle outState);

    boolean onOptionsItemSelected(MenuItem item);

    void onActivityResult(int requestCode, int resultCode, Intent data);

    void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults);

    void onRestoreInstanceState(Bundle savedInstanceState);
}

File: commons/src/main/java/com/nerdscorner/mvplib/commons/mvp/presenter/Presenter.kt
package com.nerdscorner.mvplib.commons.mvp.presenter

import android.content.Intent
import android.content.res.Configuration
import android.os.Bundle
import android.view.MenuItem

interface Presenter {

    fun onStart()

    fun onResume()

    fun onPause()

    fun onStop()

    fun onDestroyView()

    fun onBackPressed(): Boolean

    fun onConfigurationChanged(newConfig: Configuration?)

    fun onSaveInstanceState(outState: Bundle)

    fun onOptionsItemSelected(item: MenuItem?): Boolean

    fun onActivityResult(requestCode: Int, resultCode: Int, data: Intent?)

    fun onRequestPermissionsResult(requestCode: Int, permissions: Array<out String>, grantResults: IntArray)

    fun onRestoreInstanceState(savedInstanceState: Bundle?)
}
